_posts/2018-06-21-vision-for-rust-and-wasm.md

```diff
@@ -0,0 +1,255 @@
+---
+title: Our Vision for Rust and WebAssembly
+---
+
+<meta charset="utf-8" />
+
+**Surgically inserting Rust compiled to WebAssembly should be the best choice
+for speeding up the most performance-sensitive JavaScript code paths.** Do *not*
+throw away your existing code base, because Rust [plays well with
+others][better-for-all]. Regardless if you are a Rust or Web developer, your
+natural workflow shouldn't change because Rust compiled to wasm integrates
+seamlessly into your preferred tools.
+
+This blog post will expand on these aspirations and describe where we stand in
+relation to them right now. In a series of follow up posts, we will talk about
+the next steps for each major component of the Rust and WebAssembly ecosystem.
+
+<style>
+hr {
+    margin-top: 2em;
+    margin-bottom: 2em;
+}
+</style>
+
+--------------------------------------------------------------------------------
+
+> [Are you interested in helping us make these ideals into reality? Join the Rust
+and WebAssembly domain working group!][get-involved]
+
+--------------------------------------------------------------------------------
+
+## Why Focus on Performance-Sensitive Code?
+
+In the most performance-sensitive contexts, JavaScript hinders rather than
+helps. Its dynamic type system and non-deterministic garbage collection pauses
+get in the way. Seemingly small code changes can result in drastic performance
+regressions if you accidentally wander off the JIT's happy path.
+
+On the other hand, Rust gives programmers low-level control and reliable
+performance. It is free from the non-deterministic garbage collection
+pauses. Programmers have control over indirection, monomorphization, and memory
+layout.
+
+With Rust, we don't need to be a performance gurus who are intimately familiar
+with the inner workings of each JavaScript implementation's JIT. We can have
+[speed without wizardry][].
+
+## Do <u>Not</u> Rewrite — Integrate
+
+Rust compiled to WebAssembly doesn't have a runtime. This enables both small
+`.wasm` binary sizes and `.wasm` binary sizes that are proportional to the
+amount of Rust code that is being compiled to WebAssembly. Binary size is of
+huge importance since the `.wasm` must be downloaded over the network. The
+proportionality means you only pay (in code size) for what you use, and enables
+incremental and partial adoption of Rust into existing JavaScript code bases.
```

Comment:
I can't put my hand on how but I feel like we could simplify this sentence a bit